### PR TITLE
[HTTP V2] Model Configuration Support

### PR DESCRIPTION
**Pull request by CoderHam**

### CODE REVIEW

File: src/clients/python/experimental_api_v2/library/httpclient.py [hunk 1]
```diff
@@ -780,9 +780,9 @@ def unregister_cuda_shared_memory(self,
 
     def infer(self,
               inputs,
+              outputs,
```
Comment (tanmayv25) [inline src/clients/python/experimental_api_v2/library/httpclient.py]: 'outputs' is an optional field. According to specification if the 'outputs' are not provided then results for all output tensor of the model should be returned in default setting.

File: src/clients/python/experimental_api_v2/examples/simple_http_v2_infer_client.py [hunk 2]
```diff
@@ -44,8 +44,8 @@ def test_infer(model_name, input0_data, input1_data):
     outputs.append(httpclient.InferOutput('OUTPUT1'))
     query_params = {'test_1': 1, 'test_2': 2}
     results = triton_client.infer(inputs,
-                              model_name,
-                            outputs=outputs,
+                            outputs,
+                            model_name,
```
Comment (tanmayv25) [inline src/clients/python/experimental_api_v2/examples/simple_http_v2_infer_client.py]: 'outputs' should remain optional.

File: src/clients/python/experimental_api_v2/examples/simple_http_v2_class_client.py [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+    requests = []
+    results = []
+
+    # Send request
+    for inputs, outputs, FLAGS.model_name, FLAGS.model_version in requestGenerator(
+            input_name, output_name, c, h, w, format, dtype, FLAGS):
+        try:
+            results.append(
+                triton_client.infer(inputs,
+                                  outputs,
```
Comment (tanmayv25) [inline src/clients/python/experimental_api_v2/examples/simple_http_v2_class_client.py]: Here as well.

File: src/clients/python/experimental_api_v2/examples/simple_http_v2_class_client.py [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+    parser.add_argument('-u',
+                        '--url',
+                        type=str,
+                        required=False,
+                        default='localhost:8000',
+                        help='Inference server URL. Default is localhost:8000.')
+    parser.add_argument('image_filename', type=str, help='Input image.')
+    FLAGS = parser.parse_args()
+
+    # Create gRPC client for communicating with the server
```
Comment (deadeyegoodwin) [inline src/clients/python/experimental_api_v2/examples/simple_http_v2_class_client.py]: http

File: src/clients/python/experimental_api_v2/examples/simple_http_v2_class_client.py [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+
+    repeated_image_data = [image_data for _ in range(FLAGS.batch_size)]
+    batched_image_data = np.stack(repeated_image_data, axis=0)
+
+    # Set the input data
+    inputs[0].set_data_from_numpy(batched_image_data)
+
+    outputs = []
+    outputs.append(httpclient.InferOutput(output_name))
+    outputs[0].set_parameter("classification", 2)
```
Comment (deadeyegoodwin) [inline src/clients/python/experimental_api_v2/examples/simple_http_v2_class_client.py]: Why don't you use the classes argument from the command-line flags?

File: src/clients/python/experimental_api_v2/examples/simple_http_v2_class_client.py [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+
+    yield inputs, outputs, FLAGS.model_name, FLAGS.model_version
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
```
Comment (deadeyegoodwin) [inline src/clients/python/experimental_api_v2/examples/simple_http_v2_class_client.py]: This seems like the image_client example that has not really been simplified by also does not really use all the still-existing image client options? Why don't you make a full-fledged image-client example that uses the http V2 protocol? Shouldn't it mostly (completely) just be the grpc_v2_image_client.py with an option to use HTTP and a couple of minor changes? Or course rename that file to v2_image_client.py

File: src/clients/python/experimental_api_v2/library/httpclient.py (hunk 1 above)
Comment (CoderHam) [reply src/clients/python/experimental_api_v2/library/httpclient.py]: Will make these changes. Currently the server returns only outputs for the outputs mentioned in the request header

Comment (CoderHam) [reply src/clients/python/experimental_api_v2/library/httpclient.py]: Adding a ticket to track this

File: src/clients/python/experimental_api_v2/examples/simple_http_v2_class_client.py (hunk 6 above)
Comment (CoderHam) [reply src/clients/python/experimental_api_v2/examples/simple_http_v2_class_client.py]: Sure I can go ahead and do this. I was simply following the design for the simple_grpc_*_client(s). I agree that on the same basis the simple_grpc_v2_class_client can also be removed and we can have a single v2 image client as you said.

Comment (CoderHam) [reply src/clients/python/experimental_api_v2/examples/simple_http_v2_class_client.py]: Solved by #1244

File: src/clients/python/experimental_api_v2/examples/simple_grpc_v2_class_client.py [hunk 7]
```diff
@@ -176,7 +177,8 @@ def requestGenerator(input_name, output_name, c, h, w, format, dtype, FLAGS):
     inputs[0].set_data_from_numpy(batched_image_data)
 
     outputs = []
-    outputs.append(grpcclient.InferOutput(output_name, class_count=2))
+    outputs.append(grpcclient.InferOutput(output_name))
+    outputs[0].set_parameter("classification", FLAGS.classes)
```
Comment (tanmayv25) [inline src/clients/python/experimental_api_v2/examples/simple_grpc_v2_class_client.py]: There has been some changes in the library API. You must set the 'class_count' in the InferOutput constructor. See  PR #1242

File: src/clients/python/experimental_api_v2/examples/simple_http_v2_class_client.py [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+
+    repeated_image_data = [image_data for _ in range(FLAGS.batch_size)]
+    batched_image_data = np.stack(repeated_image_data, axis=0)
+
+    # Set the input data
+    inputs[0].set_data_from_numpy(batched_image_data)
+
+    outputs = []
+    outputs.append(httpclient.InferOutput(output_name))
+    outputs[0].set_parameter("classification", 2)
```
Comment (tanmayv25) [inline src/clients/python/experimental_api_v2/examples/simple_http_v2_class_client.py]: Should be set in the constructor.

Comment (CoderHam) [reply src/clients/python/experimental_api_v2/examples/simple_http_v2_class_client.py]: Done

File: src/clients/python/experimental_api_v2/examples/simple_grpc_v2_class_client.py (hunk 7 above)
Comment (CoderHam) [reply src/clients/python/experimental_api_v2/examples/simple_grpc_v2_class_client.py]: Done

File: src/clients/python/experimental_api_v2/examples/simple_grpc_v2_class_client.py [hunk 9]
```diff
@@ -32,6 +32,7 @@
 from builtins import range
```
Comment (deadeyegoodwin) [inline src/clients/python/experimental_api_v2/examples/simple_grpc_v2_class_client.py]: Should just delete these two class clients... they seem to be just be the same as or a subset of the v2_image_client.

Comment (CoderHam) [reply src/clients/python/experimental_api_v2/examples/simple_grpc_v2_class_client.py]: These two class clients verify the results of the classification are indeed correct. The v2_image_client.py simply runs the inference but does not verify the results.

Comment (deadeyegoodwin) [reply src/clients/python/experimental_api_v2/examples/simple_grpc_v2_class_client.py]: But you can/should have a CI that runs v2_image_client and verifies it's result. When we shift from V1->V2 we have several existing image_client using CI tests that will provide this coverage.

Comment (CoderHam) [reply src/clients/python/experimental_api_v2/examples/simple_grpc_v2_class_client.py]: I will make this change in the image_client PR. Including dropping the simple_grpc/http_v2_class_client.py

**Comment by CoderHam** [comment]
Thanks. Already done in the image client PR

File: src/clients/python/experimental_api_v2/examples/simple_http_v2_class_client.py [hunk 10]
```diff
@@ -0,0 +1,269 @@
...
+    repeated_image_data = [image_data for _ in range(FLAGS.batch_size)]
+    batched_image_data = np.stack(repeated_image_data, axis=0)
+
+    # Set the input data
+    inputs = []
+    inputs.append(httpclient.InferInput(input_name, batched_image_data.shape, dtype))
+    inputs[0].set_data_from_numpy(batched_image_data)
+
+    outputs = []
+    outputs.append(httpclient.InferOutput(output_name, class_count=FLAGS.classes))
```
Comment (tanmayv25) [inline src/clients/python/experimental_api_v2/examples/simple_http_v2_class_client.py:171]: You would have to use "binary_data=False" here and any 'set_data_from_numpy' calls. Have a look at #1253. Once DLIS-1204 is fixed all the example clients can enable binary data.

Comment (CoderHam) [reply src/clients/python/experimental_api_v2/examples/simple_http_v2_class_client.py]: Done

**Pull request merged**